pkgs/desktops/gnome-3/extensions/tilingnome/default.nix

```
@@ -0,0 +1,34 @@
+{ stdenv, fetchFromGitHub, glib }:
+
+stdenv.mkDerivation rec {
+  pname = "gnome-shell-extension-tilingnome";
+  version = "unstable-2019-01-18";
+
+  src = fetchFromGitHub {
+    owner = "rliang";
+    repo = "gnome-shell-extension-tilingnome";
+    rev = "bd4fb8c19f7a6282b38724b30e62645143390226";
+    sha256 = "1y4s4n88gdkpvgd3v3dg0181ccyhlixbvkx3bwyvdxyyyxbqibid";
+  };
+
+  nativeBuildInputs = [ glib ];
+
+  buildPhase = ''
+    glib-compile-schemas .
+  '';
+
+  installPhase = ''
+    mkdir -p $out/share/gnome-shell/extensions/${uuid}
+    cp -r * $out/share/gnome-shell/extensions/${uuid}/
+  '';
+
+  uuid = "tilingnome@rliang.github.com";
+
+  meta = with stdenv.lib; {
+    description = "Tiling window management for GNOME Shell";
+    license = licenses.gpl2;
+    maintainers = with maintainers; [ benley ];
+    homepage = https://github.com/rliang/gnome-shell-extension-tilingnome;
+    platforms = platforms.all;
```

```suggestion
    platforms = platforms.linux;
```

Where `platforms` was set for the gnome extensions it's this.
Also don't see it being useful outside of those platforms (where gnome is present).

---

good point, I should have looked at what gnome does.  I figured this one is just javascript so platforms don't matter much :-)

---

>  I figured this one is just javascript so platforms don't matter much

That's true, but I guess it should be matched up to where it can be deployed.